protobuf-kafkaconnect-converter/src/test/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/ProtobufSchemaConverterTest.java

```diff
@@ -151,4 +153,52 @@ public void toConnectData_convertsProtobufSerializedDataToConnectData(
         SchemaAndValue expectedSchemaAndValue = new SchemaAndValue(connectSchema, connectData);
         assertEquals(expectedSchemaAndValue, schemaAndValue);
     }
+
+    @Test
+    public void testSchemaCache_toConnectConversion() {
+        Cache<Descriptors.Descriptor, Schema> toConnectSchemaCache = protobufSchemaConverter.getToConnectSchemaCache();
+        assertEquals(0, toConnectSchemaCache.size());
+
+        Message protobufPrimitiveData = ToConnectTestDataGenerator.getPrimitiveProtobufMessages().get(0);
+        final byte[] serializedPrimitiveData = protobufPrimitiveData.toByteArray();
+        doReturn(protobufPrimitiveData).when(deserializer).deserialize(TOPIC_NAME, serializedPrimitiveData);
+        protobufSchemaConverter.toConnectData(TOPIC_NAME, serializedPrimitiveData);
+        assertEquals(1, toConnectSchemaCache.size());
+
+        //converting the same schema to see if the cache is working properly
+        doReturn(protobufPrimitiveData).when(deserializer).deserialize(TOPIC_NAME, serializedPrimitiveData);
+        protobufSchemaConverter.toConnectData(TOPIC_NAME, serializedPrimitiveData);
+        assertEquals(1, toConnectSchemaCache.size());
+
+        Message protobufEnumData = ToConnectTestDataGenerator.getEnumProtobufMessages().get(0);
+        final byte[] serializedEnumData = protobufEnumData.toByteArray();
+        doReturn(protobufEnumData).when(deserializer).deserialize(TOPIC_NAME, serializedEnumData);
+        protobufSchemaConverter.toConnectData(TOPIC_NAME, serializedEnumData);
+        assertEquals(2, toConnectSchemaCache.size());
```

Comment:
Can we also assert the contents?

Reply:
the filedescriptors do not have an equals method, so the contents are difficult to test. Manually converting and then comparing shows equal filedescriptors.